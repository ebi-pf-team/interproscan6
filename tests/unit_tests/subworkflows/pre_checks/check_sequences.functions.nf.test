
nextflow_function {

    name "Test Functions"
    script "interproscan/subworkflows/pre_checks/check_sequences.nf"

	

    test("Test Function isNucleic with nucleic sequence") {

        function "isNucleic"

        when {
            function {
                """
                input[0] = "AGCTU"
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }



    test("Test Function isNucleic with protein sequence") {

        function "isNucleic"

        when {
            function {
                """
                input[0] = "THISisAProteinSequence"
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	

    test("Test Function checkIllegalChars when no illegal chars with a protein sequence") {

        function "checkIllegalChars"

        when {
            function {
                """
                input[0] = "ADEFGHIKLMN"
                input[1] = 'antifam,sfld'.toLowerCase().split(',') as Set
                input[2] = false
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }


    test("Test Function checkIllegalChars when no illegal chars with a protein sequence but thinks its nucleic") {

        function "checkIllegalChars"

        when {
            function {
                """
                input[0] = "ADEFGHIKLMN"
                input[1] = 'antifam,sfld'.toLowerCase().split(',') as Set
                input[2] = true
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

	
    test("Test Function checkIllegalChars when there are illegal chars in a protein seq") {

        function "checkIllegalChars"

        when {
            function {
                """
                input[0] = "ADEFGHIKLMN-.,123"
                input[1] = 'antifam,sfld'.toLowerCase().split(',') as Set
                input[2] = false
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

    test("Test Function checkIllegalChars when there are illegal chars in a nucleic seq") {

        function "checkIllegalChars"

        when {
            function {
                """
                input[0] = "AGCTU-.,12_.3"
                input[1] = 'antifam,sfld'.toLowerCase().split(',') as Set
                input[2] = true
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

}
