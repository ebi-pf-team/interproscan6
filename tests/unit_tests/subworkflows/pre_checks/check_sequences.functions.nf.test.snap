{
    "Test Function isNucleic with nucleic sequence": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-28T10:49:00.398706"
    },
    "Test Function checkIllegalChars when there are illegal chars in a nucleic seq": {
        "content": [
            {
                "GENERAL": [
                    "',12_'",
                    "'3'"
                ],
                "ANTIFAM": [
                    "'-'"
                ],
                "SFLD": [
                    "'-'",
                    "'.'",
                    "'_'"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-28T10:49:14.910639"
    },
    "Test Function checkIllegalChars when no illegal chars with a protein sequence": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-28T10:49:05.915942"
    },
    "Test Function checkIllegalChars when no illegal chars with a protein sequence but thinks its nucleic": {
        "content": [
            {
                "GENERAL": [
                    "'DEF'",
                    "'HIKLMN'"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-28T10:49:08.958206"
    },
    "Test Function isNucleic with protein sequence": {
        "content": [
            false
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-28T10:49:03.107952"
    },
    "Test Function checkIllegalChars when there are illegal chars in a protein seq": {
        "content": [
            {
                "GENERAL": [
                    "',123'"
                ],
                "ANTIFAM": [
                    "'-'"
                ],
                "SFLD": [
                    "'-'",
                    "'.'"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-28T10:49:11.754557"
    }
}